extensions/wsc/src/main/java/org/mule/extension/ws/internal/WscConnectionProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.ws.internal;
+
+import org.mule.extension.ws.api.SoapVersion;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+import org.mule.runtime.api.connection.PoolingConnectionProvider;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+
+/**
+ * {@link ConnectionProvider} that returns instances of {@link WscConnection}.
+ *
+ * @since 4.0
+ */
+public class WscConnectionProvider implements PoolingConnectionProvider<WscConnection> {
```

Comment:
the pool is used to avoid parsing the same wsdl every time an operation is called, review now, I removed the capabilities to add interceptors on the fly.